### DISCUSSION TITLE
Any way to use base64 as input while storing image to Google Drive?

### DISCUSSION THREAD

**Discussion by Arturexe** [masbug/flysystem-google-drive-ext#78] [Q&A] [answered]
Storing objects from HTML input file type field to Google Drive works great. When I try to use base64 encoded (or laravel-decoded) files in the `Storage::disk('google')->put()` method, nothing happens and the file doesn't arrive in drive.

Is there any way to make it possible? Or else I guess I will have to create temporary files on my server, just to be able to send the images to drive.

**Comment by erikn69** [comment, marked as answer]
I never try that, make a test or a PR

Did you try
```php
Storage::disk('google')->put('file_name.ext', base64_decode($base64String));
```

**Reply by Arturexe** [reply]
Ok, I had the first argument of `put` specified as an empty string, which worked fine when I passed in image objects from input DOM element directly but stopped working when passing in the decoded base64 string. I've specified the file name and now it works.

**Reply by erikn69** [reply]
So, mark this as answered

**Reply by erikn69** [reply]
Input dom files have name extension and properties, base64 data has nothing, only encrypted bytes, needs a name and a extensión for processing
